src/AssetManager/View/Helper/Asset.php

```diff
@@ -0,0 +1,140 @@
+<?php
+namespace AssetManager\View\Helper;
+
+use AssetManager\Asset\AggregateAsset;
+use AssetManager\Exception\InvalidArgumentException;
+use AssetManager\Service\AssetManager;
+use Zend\ServiceManager\ServiceLocatorInterface;
+use Zend\View\Helper\AbstractHelper;
+use Zend\Cache\Storage\Adapter\AbstractAdapter as AbstractCacheAdapter;
+
+class Asset extends AbstractHelper
+{
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $serviceLocator;
+
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $assetManager;
+
+    /**
+     * @param ServiceLocatorInterface $serviceLocator
+     * @param AssetManager            $assetManager
+     * @param array                   $config
+     */
+    public function __construct(ServiceLocatorInterface $serviceLocator, AssetManager $assetManager, $config)
+    {
+        $this->serviceLocator = $serviceLocator;
+        $this->assetManager   = $assetManager;
+        $this->config         = $config;
+    }
+
+    /**
+     * find the file and if it exists, append its unix modification time to the filename
+     *
+     * @param string $filename
+     * @param string $queryString
+     * @return string
+     */
+    private function elaborateFilePath($filename, $queryString)
+    {
+        /** @var AggregateAsset $asset */
+        $asset = $this->serviceLocator->get('AssetManager\Service\AssetManager')->getResolver()->resolve($filename);
```

Comment:
Don't call `->getResolver()` on the `AssetManager`. Inject it directly instead (constructor)